plugin/src/main/java/org/opensearch/ml/searchext/MLInferenceRequestParametersUtil.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.searchext;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.search.SearchExtBuilder;
+
+public class MLInferenceRequestParametersUtil {
+
+    public static MLInferenceRequestParameters getMLInferenceRequestParameters(SearchRequest searchRequest) {
+        MLInferenceRequestParametersExtBuilder mLInferenceRequestParametersExtBuilder = null;
+        if (searchRequest.source() != null && searchRequest.source().ext() != null && !searchRequest.source().ext().isEmpty()) {
+            List<SearchExtBuilder> extBuilders = searchRequest
+                .source()
+                .ext()
+                .stream()
+                .filter(extBuilder -> MLInferenceRequestParametersExtBuilder.NAME.equals(extBuilder.getWriteableName()))
+                .collect(Collectors.toList());
+
+            if (!extBuilders.isEmpty()) {
+                mLInferenceRequestParametersExtBuilder = (MLInferenceRequestParametersExtBuilder) extBuilders.get(0);
```

Comment:
if scenario with more than one element is valid did we document current behavior of using only first element and ignoring everything else? Btw, is this guaranteed that ext elements in this list will be in the same order as in user requst?

Reply:
the extBuilders is filter by the name "ml_inference", so there will be the only type `MLInferenceRequestParametersExtBuilder`, and we are only getting the first MLInferenceRequestParametersExtBuilder. The worst case, if users use two MLInferenceRequestParametersExtBuilders, the second one is skip. We want to have one MLInferenceRequestParametersExtBuilder in every request. Hope this make sense